### PR TITLE
Add tracy.cmake

### PR DESCRIPTION
**Pull request by jorisv**
tracy.cmake generate tracy.hpp in our build directory and add PROJECT_NAME_TRACY_ENABLE cmake option.

tracy.hpp create macro that will call tracy macro/function.
When PROJECT_NAME_TRACY_ENABLE is false, tracy.hpp macro will do nothing.
This will allow to easily set/remove the tracy dependency.

### CODE REVIEW

File: tracy.cmake [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (nim65s) [inline tracy.cmake:14]: GPL ? Are you sure ?
That would be very contaminating.

Comment (jorisv) [reply tracy.cmake:14]: Sorry, I just copy past the one from header.cmake.

I will change it to the one in LICENSE.

File: tracy.hpp.in [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#define @LIBRARY_NAME@_ZONE_SCOPED ZoneScoped
+#define @LIBRARY_NAME@_ZONE_NAMED_N(varname, name, active) ZoneNamedN(varname, name, active)
+#define @LIBRARY_NAME@_ZONE_NAMED(varname, active) ZoneNamed(varname, active)
+#define @LIBRARY_NAME@_SET_THREAD_NAME(name) tracy::SetThreadName(name)
+#else
+#define @LIBRARY_NAME@_ZONE_SCOPED_N(x)
+#define @LIBRARY_NAME@_ZONE_SCOPED
+#define @LIBRARY_NAME@_ZONE_NAMED_N(x, y, z)
+#define @LIBRARY_NAME@_ZONE_NAMED(x, y)
+#define @LIBRARY_NAME@_SET_THREAD_NAME(x)
```
Comment (ManifoldFR) [inline tracy.hpp.in:30]:
```suggestion
#define @LIBRARY_NAME@_TRACY_ZONE_SCOPED_N(name) ZoneScopedN(name)
#define @LIBRARY_NAME@_TRACY_ZONE_SCOPED ZoneScoped
#define @LIBRARY_NAME@_TRACY_ZONE_NAMED_N(varname, name, active) ZoneNamedN(varname, name, active)
#define @LIBRARY_NAME@_TRACY_ZONE_NAMED(varname, active) ZoneNamed(varname, active)
#define @LIBRARY_NAME@_TRACY_SET_THREAD_NAME(name) tracy::SetThreadName(name)
#else
#define @LIBRARY_NAME@_TRACY_ZONE_SCOPED_N(x)
#define @LIBRARY_NAME@_TRACY_ZONE_SCOPED
#define @LIBRARY_NAME@_TRACY_ZONE_NAMED_N(x, y, z)
#define @LIBRARY_NAME@_TRACY_ZONE_NAMED(x, y)
#define @LIBRARY_NAME@_TRACY_SET_THREAD_NAME(x)
```

File: tracy.hpp.in
Comment (ManifoldFR) [inline tracy.hpp.in:1]: I think what we usually do is have the header template be `.cmake.in` (see `deprecated.cmake.in`)

Comment (ManifoldFR) [reply tracy.hpp.in:1]: Sorry I meant deprecated.hh.cmake. `.cmake.in` is for generating CMake module files. `tracy.hh.cmake` is the right answer!

**Review by jcarpent — APPROVED** [review]

**Comment by jorisv** [comment]
@nim65s If it's fine to you we can merge this PR.

This [aligator PR](https://github.com/Simple-Robotics/aligator/pull/191) test this new feature.

**Comment by nim65s** [comment]
Sure, this can't break anything :)

**Pull request merged**